src/proof/alethe/alethe_post_processor.h

```diff
@@ -0,0 +1,129 @@
+  /**
+   * This method updates the proof rule application by splitting on the given
+   * rule and translating it into a proof node in terms of the Alethe rules.
+   *
+   * @param res The expected result of the application,
+   * @param rule The id of the Alethe rule,
+   * @param children The children of the application,
+   * @param args The arguments of the application,
+   * @param cdp The proof to add to,
+   * @return True if the step could be added, or null if not.
+   */
+  bool update(Node res,
+              PfRule id,
+              const std::vector<Node>& children,
+              const std::vector<Node>& args,
+              CDProof* cdp,
+              bool& continueUpdate) override;
+
+ private:
+  /** The proof node manager */
+  ProofNodeManager* d_pnm;
+  /** The node manager */
+  NodeManager* d_nm;
+  /** The variable cl **/
+  Node d_cl;
+  /**
+   * This method adds a new step to the proof applying the ALETHE_RULE. It adds
+   * the id of the ALETHE_RULE as the first argument, the res node as the second
+   * and third argument.
+   *
+   * @param res The expected result of the application,
+   * @param rule The id of the Alethe rule,
+   * @param children The children of the application,
+   * @param args The arguments of the application,
+   * @param cdp The proof to add to,
+   * @return True if the step could be added, or false if not.
+   */
+  bool addAletheStep(Node res,
+                    AletheRule rule,
+                    const std::vector<Node>& children,
+                    const std::vector<Node>& args,
+                    CDProof& cdp);
+  /**
+   * This method adds a new step to the proof applying the ALETHE_RULE but adds
+   * a conclusion different from the result as the third argument.
+   *
+   * @param res The expected result of the application,
+   * @param rule The id of the Alethe rule,
+   * @param conclusion The conclusion of the application as the Alethe printer
+   * @param children The children of the application,
+   * @param args The arguments of the application
+   * @param cdp The proof to add to
+   * @return True if the step could be added, or false if not.
+   */
+  bool addAletheStep(Node res,
+                    AletheRule rule,
+                    Node conclusion,
+                    const std::vector<Node>& children,
+                    const std::vector<Node>& args,
+                    CDProof& cdp);
+  /**
+   * This method adds a new step to the proof applying the Alethe rule while
+   * replacing the outermost or by cl, i.e. (cl F1 ... Fn). The kind of the
+   * given Node has to be OR.
+   *
+   * @param res The expected result of the application in form (or F1 ... Fn),
+   * @param rule The id of the Alethe rule,
+   * @param children The children of the application,
+   * @param args The arguments of the application
+   * @param cdp The proof to add to
+   * @return True if the step could be added, or false if not.
+   */
+  bool addAletheStepFromOr(Node res,
```

Comment:
Do I understand correctly that this method is taking `(or F1 ... Fn)` and effectively calling the version above with `(or F1 ... Fn)` as the first argument and `(cl F1 ... Fn)` as the third? 

I think it's fine to keep this method, but explain it in terms of the above method to have things more clear.

Reply:
Yes, exactly. I will make this clearer in the comments.